astropy/table/table.py

```diff
@@ -1441,6 +1436,18 @@ def dtype(self):
     def colnames(self):
         return list(self.columns.keys())
 
+    def _is_list_or_tuple_of_colnames(self, names):
+        """Check that ``names`` is a valid tuple or list for column-slicing self"""
+        if (isinstance(names, (tuple, list)) and names and
+                all(isinstance(x, str) for x in names)):
+            bad_names = [x for x in names if x not in self.colnames]
+            if bad_names:
+                raise ValueError('Slice name(s) {0} not valid column name(s)'
```

Comment:
I guess this is too late, but shouldn't this be `KeyError`? That is what one gets with a non-existing single column name. Note that if one were to just remove this test, it would give the `KeyError` later anyway.

Reply:
Yup, `KeyError` would make more sense.  Sigh.